packages/tokens/non-fungible/src/extensions/consecutive/storage.rs

```diff
@@ -369,8 +405,10 @@ impl Consecutive {
             let approval_key = StorageKey::Approval(token_id);
             e.storage().temporary().remove(&approval_key);
 
-            // Set the next token to prev owner
-            Consecutive::set_owner_for(e, from_address, token_id + 1);
+            // Set the previous token to prev owner, if there's a previous id.
+            if token_id > 0 {
+                Consecutive::set_owner_for(e, from_address, token_id - 1);
```

Comment:
why don't we call:
```
e.storage().persistent().set(&StorageKey::Owner(token_id), to_address);
Self::set_ownership_in_bucket(e, token_id);
```

as in below?

Reply:
because we need to check the previous (`token_id - 1`) is a "free" token (not burned or not owned by someone else) and if it's the case, `set_owner_for` does exactly this

Reply:
ok great, can we also document this as inline comments? If I couldn't understand it, probably other people will have similar problems